tests/unit/test_openapi/test_schema.py

```diff
@@ -397,3 +401,39 @@ def test_schema_generation_with_pagination(annotation: Any) -> None:
     assert properties["foo"] == expected_foo_schema
     assert properties["annotated_foo"] == expected_foo_schema
     assert properties["optional_foo"] == expected_optional_foo_schema
+
+
+def test_optional_enum() -> None:
+    class Foo(Enum):
+        A = 1
+        B = 2
+
+    schema = SchemaCreator().for_field_definition(FieldDefinition.from_annotation(Optional[Foo]))
+    assert isinstance(schema, Schema)
+    assert schema.type == OpenAPIType.INTEGER
+    assert schema.enum == [1, 2, None]
+
+
+def test_optional_literal() -> None:
+    schema = SchemaCreator().for_field_definition(FieldDefinition.from_annotation(Optional[Literal[1]]))
+    assert isinstance(schema, Schema)
+    assert schema.type == OpenAPIType.INTEGER
+    assert schema.enum == [1, None]
+
+
+@pytest.mark.parametrize(
+    ("in_type", "out_type"),
+    [
+        (FieldDefinition.from_annotation(Optional[int]), int),
+        (FieldDefinition.from_annotation(Union[None, int]), int),
+        (FieldDefinition.from_annotation(int), int),
+        # hack to create a union of NoneType, NoneType to hit a branch for coverage
+        (FieldDefinition.from_annotation(_GenericAlias(Union, (NoneType, NoneType))), ValueError),
```

Comment:
This is the dark side of branch coverage.. the only way that the condition of exhausting the generator in the function under test could be hit is if there is no non-None type in the union with None. Not possible in the real world, but here we are.

Reply:
To preoccupied to see if I could, didn't stop to think if I should.. etc etc.